### ISSUE TITLE
Use `id` instead of `index`

### ISSUE THREAD

**Issue by khernyo** [datrs/flat-tree#30]
It seems to me that the data referred to as `index` is not really an index. At least it's not the node index when building the merkle-tree. It's a bit confusing when the nodes are stored in a flat data structure (e.g., a list).

When building the merkle-tree, there is an inherent ordering which is the order in which each hash is computed (using this order requires the least amount of memory to build the merkle-tree):
1. first data block is received, its hash is computed (A)
1. second data block is received, its hash is computed (B)
1. hash of (AB) is computed (C)
1. third data block is received, hashes computed (D)
1. fourth data block is received, hashes computed (E)
1. hash of (DE) is computed (F)
1. hash of (CF) is computed (G)
...

The `index` for the above hashes in the flat-tree are the following:
- A: 0
- B: 2
- C: 1
- D: 4
- E: 6
- F: 5
- G: 3

Maybe there is a reason for using `index` to refer to these nodes, but I could not find one. So, if there is no specific reason for this name, I propose to use `id` or maybe `position` instead.

**Comment by yoshuawuyts** [comment]
In the case of this module I've mostly followed the prior art at https://github.com/mafintosh/flat-tree. I see where you're coming from, but mostly for compat reasons I reckon it might be best to keep things similar.

I personally don't care too much about naming in this case, as probably the best approach is to keep things similar so bridging between the two is as straight forward as possible.

Hope that makes sense!

**Comment by khernyo** [comment]
Yes, the terminology should be in sync, so I opened the same issue in that repo too: https://github.com/mafintosh/flat-tree/issues/12

I assume there is no other reason against the rename.
